### PR TITLE
Create composite action for Dispatch Build Bottle build job

### PR DESCRIPTION
**Pull request by GauthamGoli**
https://github.com/Homebrew/brew/issues/11101

### CODE REVIEW

File: dispatch-build-bottle/action.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        echo "::set-output name=repo::$(brew --repo ${{github.repository}})"
+
+    - name: Set up Homebrew
+      id: set-up-homebrew
+      uses: Homebrew/actions/setup-homebrew@master
+
+    - name: Run brew test-bot --only-cleanup-before
+      if: github.repository == 'Homebrew/homebrew-core'
+      id: test-bot-only-cleanup-before
+      run: brew test-bot --only-cleanup-before
```
Comment (GauthamGoli) [inline dispatch-build-bottle/action.yml:46]: Steps which have  `if: github.repository == 'Homebrew/homebrew-core'`, were present only in `homebrew-core`'s `dispatch-build-bottle` action,  but not for `linuxbrew-core` 

Can this condition be modified to `if: always()` ?  I'm not sure and need input here

File: dispatch-build-bottle/action.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      if: always()
+      env:
+        HOMEBREW_GITHUB_API_TOKEN: ${{secrets.GITHUB_TOKEN}}
+      run: |
+        mkdir ~/bottles
+        cd ~/bottles
+        brew test-bot \
+          --only-formulae \
+          --keep-old \
+          --skip-recursive-dependents \
```
Comment (GauthamGoli) [inline dispatch-build-bottle/action.yml:63]: `--skip-recursive-dependents` is not present in `homebrew-core`'s action's `step` ,
Does this step need to be separated  based on `github.repository` ?

**Comment by dawidd6** [comment]
Did you test this Action? Cause I can tell right now that it might not work in current form. AFAIK composite Actions does not support running other Actions using `uses` keyword. Did something change recently around in this matter?

**Comment by dawidd6** [comment]
```
Error: GauthamGoli/actions/dispatch-build-bottle-composite/dispatch-build-bottle/action.yml (Line: 27, Col: 7): Required property is missing: shell
Error: GauthamGoli/actions/dispatch-build-bottle-composite/dispatch-build-bottle/action.yml (Line: 41, Col: 7): Unexpected value 'uses'
...
```

Idea is good, but I don't think it will work (for now).

File: dispatch-build-bottle/action.yml (hunk 1 above)
Comment (RandomDSdevel) [reply dispatch-build-bottle/action.yml:46]: &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;No, some users might (well?) then run into issues like the ones I and @reitermarkus fixed in Homebrew/brew#9457, Homebrew/homebrew-core#65917, and Homebrew/homebrew-cask#89714.  You can still loosen this condition to only run actions if they're triggered from within the Homebrew organization, though.

**Comment by GauthamGoli** [comment]
As noted by @dawidd6  this will not work for now. More details are posted in the above linked parent issue on why this will not work.

**Pull request closed without merging**